zarr/tests/test_storage.py

```diff
@@ -1549,6 +1549,13 @@ def test_permissions(self):
             assert perm == '0o40775'
         z.close()
 
+    def test_store_and_retrieve_ndarray(self):
+        store = ZipStore('data/store.zip')
```

Comment:
I'm not actually sure why some of these tests use `data/store.zip` explicitly here, but I think for your test, @orenwatson, it would be best to use the `self.create_store()` method so that you are testing a fresh zip, independently of other tests.

I can confirm though that this test fails without this PR and passes with it. :+1:

NB: A _second_ run with this PR raises the warning:

```
/usr/local/anaconda3/envs/z/lib/python3.9/zipfile.py:1505: UserWarning: Duplicate name: 'foo'
  return self._open_to_write(zinfo, force_zip64=force_zip64)
```

which is what makes me think that `create_store` would be safer.

Reply:
Good point Josh 👍

I think with `test_permissions` this is needed because the file needs to be flushed/closed and then reopened for the test (so it doesn't follow the norm). Though agree we should be able to use `create_store` here

Reply:
Makes sense. If the test is passing, I say we release @orenwatson (and add a grand "Huzzah!" in thanks 🎉) and we can handle the cleanup separately.